lighthouse-core/report/v2/renderer/dom.js

```diff
@@ -37,7 +37,9 @@ class DOM {
       element.className = className;
     }
     Object.keys(attrs).forEach(key => {
-      element.setAttribute(key, attrs[key]);
+      if (attrs[key] !== undefined) {
```

Comment:
Every caller will have to do that. I'd rather just prevent `attr="undefined"` in one place. Kinda what templating languages do for you ;)